src/qt/transactionrecord.cpp

```diff
@@ -312,56 +365,15 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet*
             //
             // Debit
             //
-            CAmount nTxFee = nDebit - wtx.GetValueOut();
-
-            for (unsigned int nOut = 0; nOut < wtx.vout.size(); nOut++) {
-                const CTxOut& txout = wtx.vout[nOut];
-                TransactionRecord sub(hash, nTime, wtx.GetTotalSize());
-                sub.idx = (int) nOut;
-                sub.involvesWatchAddress = involvesWatchAddress;
-
-                if (wallet->IsMine(txout)) {
-                    // Ignore parts sent to self, as this is usually the change
-                    // from a transaction sent back to our own address.
-                    continue;
-                }
-
-                CTxDestination address;
-                if (ExtractDestination(txout.scriptPubKey, address)) {
-                    //This is most likely only going to happen when resyncing deterministic wallet without the knowledge of the
-                    //private keys that the change was sent to. Do not display a "sent to" here.
-                    if (wtx.HasZerocoinMintOutputs())
-                        continue;
-                    // Sent to PIVX Address
-                    sub.type = TransactionRecord::SendToAddress;
-                    sub.address = EncodeDestination(address);
-                } else if (txout.IsZerocoinMint()){
-                    sub.type = TransactionRecord::ZerocoinMint;
-                    sub.address = getValueOrReturnEmpty(wtx.mapValue, "zerocoinmint");
-                    sub.credit += txout.nValue;
-                } else {
-                    // Sent to IP, or other non-address transaction like OP_EVAL
-                    sub.type = TransactionRecord::SendToOther;
-                    sub.address = getValueOrReturnEmpty(wtx.mapValue, "to");
-                }
-
-                CAmount nValue = txout.nValue;
-                /* Add fee to first output */
-                if (nTxFee > 0) {
-                    nValue += nTxFee;
-                    nTxFee = 0;
-                }
-                sub.debit = -nValue;
-
-                parts.append(sub);
-            }
-        } else {
-            //
-            // Mixed debit transaction, can't break down payees
-            //
-            parts.append(TransactionRecord(hash, nTime, wtx.GetTotalSize(), TransactionRecord::Other, "", nNet, 0));
-            parts.last().involvesWatchAddress = involvesWatchAddress;
+            decomposeDebitTransaction(wallet, wtx, nDebit, involvesWatchAddress, parts);
+            return parts;
```

Comment:
(e6c637960177e601b9d67b707a90e7a6f68f254a) the return value of `decomposeDebitTransaction` is always true, and it never gets used. 
Same for `decomposeSendToSelfTransaction` (in 5e894df145fb16785c8f43342f88e2d53d24ecfd) and `decomposeCreditTransaction` (in 162efba46715dfdfc516351530c4fbda9ea842b1).

I think we should either keep (for consistency among all "decompose" functions)
```cpp
if (decomposeDebitTransaction(wallet, wtx, nDebit, involvesWatchAddress, parts))
    return parts;
```
or just make them `void` functions.

Reply:
yep, sounds good.